protocol/x/clob/abci.go

```diff
@@ -238,12 +238,24 @@ func PrepareCheckState(
 		if optimisticallyFilledQuantums > 0 {
 			numFilledLiquidations++
 		} else {
-			telemetry.IncrCounter(
-				1,
-				types.ModuleName,
-				metrics.PrepareCheckState,
-				metrics.UnfilledLiquidationOrders,
-			)
+			telemetry.IncrCounter(1, types.ModuleName, metrics.PrepareCheckState, metrics.UnfilledLiquidationOrders)
+
+			// The liquidation order was unfilled. Try to deleverage the subaccount.
+			subaccountId := liquidationOrders[i].GetSubaccountId()
+			perpetualId := liquidationOrders[i].MustGetLiquidatedPerpetualId()
+			deltaQuantums := liquidationOrders[i].GetDeltaQuantums()
+
+			_, err := keeper.DeleverageSubaccount(ctx, subaccountId, perpetualId, deltaQuantums)
+			if err != nil {
```

Comment:
Since this function cannot return nil. I think the only error that is possible to trigger this is when subaccount cannot be deleveraged (doesn't meet the two conditions). 

Should we stat here?

Reply:
when subaccounts cannot be deleveraged, error will be nil.

error will be non nil if `GetNetCollateralAndMarginRequirements` returns an error, and this should never happen.